crates/cow-amm/src/amm.rs

```diff
@@ -23,8 +24,10 @@ impl Amm {
         helper: &CowAmmLegacyHelper,
     ) -> Result<Self, MethodError> {
         let tradeable_tokens = helper.tokens(address).call().await?;
+        let contract = contracts::CowAmm::at(&helper.raw_instance().web3(), address);
```

Comment:
nit: I think this should be passed in from above (instead of now having a contract and an address field on the struct which are both implicitely the same thing).